Update list to include SS4 modules and modules in CWP 2.0 recipes

raissanorth opened the pull request:
Also:

- introduce two new boolean attributes, scrutinizer and addons, to aid in displaying the table of modules
- delete uncommented code. 
- link to addons page, rather than GitHub
- update links of repos that moved from GitLab to GitHub.
- update references to silverstripe-australia, since renaming to symbiote
- retrieve badges directly from Scrutinizer where possible and display custom shield badge, where not possible

Fixes https://github.com/silverstripeltd/cc-issues/issues/124

chillu commented on modules.json:
```diff
@@ -1,362 +1,963 @@
+        "type": "supported-module"
+    },
+    {
+        "github": "bringyourownideas\/silverstripe-composer-update-checker",
```
Should those just be supported dependencies, with `silverstripe-maintanence` as the supported module? This would allow us to provide the same functionality in a different way

chillu commented on modules.json:
```diff
@@ -1,362 +1,963 @@
         "type": "supported-dependency"
     },
     {
-        "github": "silverstripe-australia\/silverstripe-gridfieldextensions",
```
It's a bit hard to read this diff with both edits and ordering changes in the same commit (`gridfieldextensions` is reintroduced later on in the diff)

chillu commented on modules.json:
```diff
@@ -1,362 +1,963 @@
+        "type": "supported-module"
+    },
+    {
+        "github": "silverstripe\/silverstripe-dms-cart",
```
I guess we don't have a way to qualify "supported under CWP 1.x", right? It'll become clear by the lack of available CWP 2.x compatible releases

raissanorth replied:
We could, but we'd have to add yet another attribute to the list.

robbieaverill replied:
We thought about doing this with semver constraints and dates and stuff but decided to go with a boolean 😆

chillu commented on modules.json:
```diff
@@ -1,362 +1,963 @@
+        "type": "supported-module"
+    },
+    {
+        "github": "silverstripe\/silverstripe-graphql-devtools",
```
Why does this need to be supported?

chillu commented on modules.json:
```diff
@@ -1,362 +1,963 @@
+        "type": "supported-module"
+    },
+    {
+        "github": "silverstripe\/silverstripe-moduleratings",
```
I don't this `silverstripe-moduleratings` or `silverstripe-moduleratings-plugin` need to be declared as supported

chillu commented on modules.json:
```diff
@@ -1,362 +1,963 @@
     },
     {
-        "github": "silverstripe-australia\/silverstripe-multivaluefield",
+        "github": "silverstripe\/silverstripe-mssql",
```
Nope, please don't add this.

chillu commented:
This is adding a whole bunch of stuff to supported modules - some of them are obvious, some are judgement calls which should discuss. That's hard to do with this diff. @raissanorth Can you please list the modules which have been added and removed here? Alternatively, separate commits for ordering changes and add/remove changes :)

raissanorth commented:
Sorry about that. Given that this repo is used to generate the list of commercially supported modules on https://www.silverstripe.org/software/addons/silverstripe-commercially-supported-module-list/, I'll just focus on the changes made to the latest version of that page.

*Modules added:*

```
bringyourownideas/silverstripe-maintenance			
bringyourownideas/silverstripe-composer-update-checker			
bringyourownideas/silverstripe-composer-security-checker	
composer/installers				
cwp/cwp-recipe-cms	
cwp/cwp-recipe-core		
dnadesign/silverstripe-elemental			
dnadesign/silverstripe-elemental-subsites			
dnadesign/silverstripe-elemental-userforms	
cwp/cwp-installer	
lekoala/silverstripe-debugbar				
silverstripe/activedirectory				
silverstripe/admin	
silverstripe/asset-admin		
silverstripe/assets	
silverstripe/behat-extension				
silverstripe/config	
silverstripe/controllerpolicy		
silverstripe/elemental-blocks		
silverstripe/eslint-config			
silverstripe/gridfieldqueuedexport	
silverstripe/moduleratings		
silverstripe/moduleratings-plugin			
silverstripe/mssql
silverstripe/postgresql			
silverstripe/raygun
silverstripe/realme		
silverstripe/recipe-cms				
silverstripe/recipe-content-blocks		
silverstripe/recipe-core
silverstripe/recipe-plugin
silverstripe/upgrader		
silverstripe/versioned			
silverstripe/webpack-config			
silverstripe-themes/simple	
tractorcow/silverstripe-proxy-db		
```

*Modules Renamed:*

```
silverstripe/cwp-recipe-search into cwp/cwp-recipe-search
```

*Included Modules That Were Already Listed on [silverstripe.org](https://www.silverstripe.org/software/addons/silverstripe-commercially-supported-module-list/):*

```
colymba/gridfield-bulk-editing-tools	
cwp/agency-extensions	
cwp/cwp		
cwp/cwp-core	
cwp/cwp-pdfexport	
cwp/cwp-recipe-basic
cwp/cwp-recipe-basic-dev
cwp/cwp-recipe-blog	
cwp/cwp-search	
cwp/starter-theme	
cwp/watea-theme
cwp-themes/default	
hafriedlander/phockito	
hafriedlander/silverstripe-phockito	
silverstripe/akismet	
silverstripe/auditor		
silverstripe/blog	
silverstripe/campaign-admin			
silverstripe/cms	
silverstripe/comment-notifications			
silverstripe/comments		
silverstripe/content-widget	
silverstripe/contentreview	
silverstripe/crontask			
silverstripe/dms			
silverstripe/dms-cart			
silverstripe/documentconverter		
silverstripe/environmentcheck		
silverstripe/errorpage				
silverstripe/externallinks			
silverstripe/framework		
silverstripe/fulltextsearch		
silverstripe/graphql		
silverstripe/graphql-devtool			
silverstripe/html5	
silverstripe/hybridsessions		
silverstripe/iframe	
silverstripe/installer	
silverstripe/ldap	
silverstripe/lumberjack	
silverstripe/mimevalidator		
silverstripe/recipe-authoring-tools			
silverstripe/recipe-blog				
silverstripe/recipe-collaboration		
silverstripe/recipe-form-building		
silverstripe/recipe-reporting-tools		
silverstripe/recipe-services	
silverstripe/registry		
silverstripe/reports		
silverstripe/restfulserver				
silverstripe/secureassets				
silverstripe/securityreport				
silverstripe/segment-field		
silverstripe/selectupload				
silverstripe/sharedraftcontent				
silverstripe/siteconfig		
silverstripe/sitewidecontent-report		
silverstripe/spamprotection				
silverstripe/spellcheck			
silverstripe/sqlite3				
silverstripe/sspak			
silverstripe/staticpublisher				
silverstripe/staticpublishqueue			
silverstripe/subsites			
silverstripe/tagfield			
silverstripe/taxonomy			
silverstripe/textextraction		
silverstripe/translatable
silverstripe/vendor-plugin			
silverstripe/versioned-admin		
silverstripe/versionfeed	
silverstripe/widgets	
symbiote/silverstripe-advancedworkflow		
symbiote/silverstripe-gridfieldextensions	
symbiote/silverstripe-multivaluefield	
symbiote/silverstripe-queuedjobs
symbiote/silverstripe-versionedfiles		
tijsverkoyen/akismet			
tractorcow/classproxy			
tractorcow/silverstripe-fluent		
undefinedoffset/sortablegridfield
```

chillu commented:
I've read through https://www.silverstripe.org/software/addons/supported-modules-definition/, and would suggest some changes to that list based on it. In particular, reclassifying some of this as a "supported dependency" with lessened expectations on our maintenance efforts. Maybe that's the case for some of the below already, can't tell from the list :)

 * silverstripe/behat-extension: Should become a "supported dependency". It's one way through which we perform testing, and while it's heavily used in core, it's not had much adoption beyond that. There's a lot of other ways to do this in the wider ecosystem. I don't want this to tie us down if we decide to switch to one of those in core.
 * silverstripe/moduleratings and silverstripe/moduleratings-plugin: Those are means to an end (site summariser), so I'd say they're a "supported dependency".
 * silverstripe/mssql: That shouldn't be on this list, it's not part of any of our QA processes, and we actively steer people away from using it
 * silverstripe/raygun: It's useful to SilverStripe Ltd, but I don't see why it needs to be supported beyond that
 * silverstripe/webpack-config: That's an internal tool to avoid repetition, "supported dependency" at best
 * tractorcow/classproxy and tractorcow/silverstripe-proxy-db: Not sure why those are on the list?
 * silverstripe/graphql-devtool: Make this a "supported dependency"
 * silverstripe/staticpublisher: We aren't really supporting this, and steer people towards the (actually supported) silverstripe/staticpublishqueue. Is it time to stop supporting this officially?

robbieaverill commented:
>tractorcow/classproxy and tractorcow/silverstripe-proxy-db: Not sure why those are on the list?

They're dependencies of silverstripe/fulltextsearch, silverstripe/auditor and lekoala/silverstripe-debugbar

raissanorth commented:
@chillu Thanks for your feedback. 

I have changed the `behat-extension`, `moduleratings`, `moduleratings-plugin`, `graphql-devtools` to supported dependencies.

I have removed `mssql`, `webpack-config`, `staticpublisher` and `raygun` from the list.

I added`proxy-db` as a supported dependency because a number of supported modules depend on it, e.g. auditor and fulltextsearch. `classproxy` is a dependency for `proxy-db` . @tractorcow What are your plans for the future of these? Are they gonna go into core?

raissanorth commented:
I published the latest version of the list to https://www.silverstripe.org/software/addons/silverstripe-commercially-supported-module-list/
Could you please sanity check @be2n @brynwhyman, thanks!

tractorcow commented:
I'm hoping that by 5.0 we'll have a DB API that lets us extend the database classes properly without needing a proxy. I'll support the proxy modules until that time.

be2n commented:
Would it make sense to have a timestamp on when the list was last updated?

chillu commented:
Lookin' good! One question: Why is `dnadesign/silverstripe-elemental` a supported dependency rather than a supported module?

raissanorth commented:
Good ideas! I added a "timestamp" below the list featured on https://www.silverstripe.org/software/addons/silverstripe-commercially-supported-module-list/. 

If there is a better way to do this within the CMS, please let me know.

![image](https://user-images.githubusercontent.com/14869519/41256861-f8f37568-6e1e-11e8-93d0-025649880e8f.png)

raissanorth commented:
@chillu, nice spotting! I will update `dnadesign/silverstripe-elemental` to a supported module.
